src/mainwindow.cpp

```diff
@@ -125,22 +125,43 @@ QString MainWindow::runTranslation(API *api,
                                    const QString &sourceLanguage,
                                    const QString &targetLanguage) {
     if (api == nullptr) {
+        showErrorBox("Text cannot be translated. API is not set.");
         return QString();
     }
 
-    QString outputString = api->translate(inputString, sourceLanguage, targetLanguage);
+    QString outputString;
+    try {
+        outputString = api->translate(inputString, sourceLanguage, targetLanguage);
+    } catch (const InvalidArgumentException &e) {
+        showErrorBox("Text cannot be translated. Invalid API key.");
```

Comment:
Use message added into `InvalidArgumentException` and add it to `showErrorBox` as additional details.

Same with other instances of this exception.

Reply:
I did a bit in a different way,